src/pq_flash_index.cpp

```diff
@@ -1333,24 +1328,18 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
             uint32_t *node_buf = OFFSET_TO_NODE_NHOOD(node_disk_buf);
             uint64_t nnbrs = (uint64_t)(*node_buf);
             T *node_fp_coords = OFFSET_TO_NODE_COORDS(node_disk_buf);
-            //        assert(data_buf_idx < MAX_N_CMPS);
-            if (data_buf_idx == MAX_N_CMPS)
-                data_buf_idx = 0;
 
-            T *node_fp_coords_copy = data_buf + (data_buf_idx * aligned_dim);
-            data_buf_idx++;
-            memcpy(node_fp_coords_copy, node_fp_coords, disk_bytes_per_point);
             float cur_expanded_dist;
             if (!use_disk_index_pq)
             {
-                cur_expanded_dist = dist_cmp->compare(aligned_query_T, node_fp_coords_copy, (uint32_t)aligned_dim);
+                cur_expanded_dist = dist_cmp->compare(aligned_query_T, node_fp_coords, (uint32_t)aligned_dim);
```

Comment:
I do not think we need to copy the data to some buf, then compute the distance. We do not compute the distance in-parallel, so I do not think we need to use "coord_scratch". Maybe some old version uses this way to compute the distance in-parallel